go-controller/pkg/clustermanager/egressip_event_handler.go

```diff
@@ -133,16 +132,12 @@ func (h *egressIPClusterControllerEventHandler) UpdateResource(oldObj, newObj in
 			}
 			return nil
 		}
-		if isOldReady == isNewReady && !isHostCIDRsAltered {
-			return nil
-		}
 		if !isNewReady {
 			klog.Warningf("Node: %s is not ready, deleting it from egress assignment", newNode.Name)
 			if err := h.eIPC.deleteEgressNode(newNode.Name); err != nil {
 				return err
 			}
 		} else if isNewReady && isNewReachable {
-			klog.Infof("Node: %s is ready and reachable, adding it for egress assignment", newNode.Name)
 			h.eIPC.setNodeEgressReachable(newNode.Name, isNewReachable)
 			if err := h.eIPC.addEgressNode(newNode.Name); err != nil {
```

Comment:
NIT: `addEgressNode` starts with:
```
klog.V(5).Infof("Egress node: %s about to be initialized", nodeName)
```
I think it would be worth moving it somewhere where it actually does anything, like under the condition to reconcile the eip:
```
	for _, egressIP := range egressIPs {
		egressIP := *egressIP
		if len(egressIP.Spec.EgressIPs) != len(egressIP.Status.Items) {
```

Reply:
ack that's a good point, let me fix that here.